### DISCUSSION TITLE
依赖cryptopp找不到头文件

### DISCUSSION THREAD

**Discussion by xjavr9** [xmake-io/xmake#3632] [General]
新手遇到问题望解答

我的xmake.lua如下

add_requires("cryptopp")
add_rules("mode.debug", "mode.release")
target("test")
    set_kind("binary")
    add_files("src/*.cpp")

我的cpp文件如下
#include <iostream>
#include <cryptopp/cryptlib.h>
#include <cryptopp/hex.h>

using namespace std;

int main(int argc, char** argv)
{
    cout << "hello world!" << endl;
    return 0;
}

PS E:\dWork\test\xmake\test> xmake
checking for Microsoft Visual Studio (x64) version ... 2019
[ 25%]: compiling.release src\main.cpp
error: main.cpp
src\main.cpp(2): fatal error C1083: 无法打开包括文件: “cryptopp/cryptlib.h”: No such file or directory

xmake编译提示找不到头文件，我觉得很奇怪，因为代码cryptopp的库已经下载下来了，路径上也有，头文件也有

C:\Users\Administrator\AppData\Local\.xmake\packages\c\cryptopp\8.7.0\a8c8af43ddfd46a2aff900c28da0b6d7\
这个路径已经有文件
bin/
include/
lib/
share/
manifest.txt
references.txt

其中include/cryptopp/cryptlib.h 这个文件是存在的

现在不知道怎么整了。

**Comment by star-hengxing** [comment]
没加 add_packages？

**Reply by xjavr9** [reply]
成功了，还有这个操作，刚刚用不明白为啥

**Reply by xjavr9** [reply]
好像还是不行，刚刚执行你的命令以为成功了

**Comment by waruqi** [comment]
target 里面缺少 `add_packages("cryptopp")`

**Reply by xjavr9** [reply]
这回是成功了，谢谢，我以为依赖加add_requires("cryptopp")就好了

**Reply by wanzzhehe** [reply]
> 这回是成功了，谢谢，我以为依赖加add_requires("cryptopp")就好了

提个建议，以后粘贴代码、文本的时候可以用代码块功能，

用法如下：

\`\`\`[language name: text, c, cpp, python, etc.]

code here

\`\`\`

效果：

```cpp
#include <iostream>

int main() {
    std::cout << "Hello world!" << std::endl;
    return 0;
}
```
